File: LeetCode/Easy/14. Longest Common Prefix.java
import java.util.*;
import java.io.*;

class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        if (strs.length == 0) {
            return "";
        }
        
        String start = strs[0];
        
        for (int i = 1; i < strs.length; i++)
            
            while (strs[i].indexOf(start) != 0) {
                
                start = start.substring(0, start.length() - 1);
                
                if (start.isEmpty()) {
                    return "";
                }
            }  
        
        return start;
        
    }
    
}

File: LeetCode/Python/Easy/14. Longest Common Prefix.py
"""
Write a function to find the longest common prefix string amongst an array of strings.

If there is no common prefix, return an empty string "".
"""
class Solution(object):
    def longestCommonPrefix(self, strs):
        output = ""
        pos = 0
        min_len = 0
        for x in range(len(strs)):
            if x == 0:
                min_len = len(strs[0])
            else:
                if len(strs[x]) < min_len:
                    min_len = len(strs[x])

        for x in range(min_len):
            val = strs[0][pos]

            for word in strs:
                if word[pos] != val:
                    return output

            output += val
            pos += 1

        return output
